.github/workflows/hw_mode_test.yml

```diff
@@ -948,3 +948,63 @@ jobs:
       run: |
         docker exec ${{ env.CONTAINER_NAME }} bash -c "cat /root/occlum/build/test/.fail"
         docker stop ${{ env.CONTAINER_NAME }}
+  
+  Spark_test:
+    if: github.event_name == 'push' || ${{ contains(github.event.pull_request.labels.*.name, 'SGX-hardware-test-required') }}
+    runs-on: ${{ matrix.self_runner }}
+    strategy:
+      matrix:
+        self_runner: [[self-hosted, SGX2-HW]]
+
+    steps:
+    - name: Clean before running
+      run: |
+        sudo chown -R ${{ secrets.CI_ADMIN }} "${{ github.workspace }}"
+
+    - name: Checkout code
+      if: github.event_name == 'push'
+      uses: actions/checkout@v2
+      with:
+        submodules: true
+
+    - name: Checkout code from fork
+      if: ${{ contains(github.event.pull_request.labels.*.name, 'SGX-hardware-test-required') }}
+      uses: actions/checkout@v2
+      with:
+        ref: refs/pull/${{ github.event.pull_request.number }}/merge
+        submodules: true
+    
+    - name: Create container
```

Comment:
It would be better to utilize the HW composite action `"- uses: ./.github/workflows/composite_action/hw"`.
For the special docker run flags "-v $(which docker):/usr/bin/docker -v /var/run/docker.sock:/var/run/docker.sock", I think you can refer to [NGO SIM composite action](https://github.com/occlum/ngo/blob/master/.github/workflows/composite_action/sim/action.yml#L15) to add similar ones for Occlum as well.

Reply:
Done, nice suggestion!